tests/custom/test_timeseries_preprocessing.py

```diff
@@ -0,0 +1,193 @@
+from unittest import TestCase
+
+import numpy as np
+import pandas as pd
+from numpy.testing import assert_allclose
+
+from mlprimitives.custom.timeseries_preprocessing import (
+    intervals_to_mask, rolling_window_sequences, time_segments_aggregate, time_segments_average)
+
+
+class IntervalsToMaskTest(TestCase):
```

Comment:
Can we add the following test cases?
- Intervals is an empty list or numpy array
- An interval starts or ends exactly on one of the index ends.
- The interval covers the whole index: `intervals = [[index[0], index[-1]]]`
- Some interval starts or ends before/after the index: `index = [3, 4, 5], intervals = [[1, 4]]`